### PR TITLE
feat: Expressify `str.slice`

### PR DESCRIPTION
**Pull request by reswqa**
This comes from @cmdlineluser's nice work. It's hard to make any changes in the original commit due to too many code conflicts, but I'm listing him as a co-author to keep authorship :)

### CODE REVIEW

**Comment by alexander-beedie** [comment]
> @alexander-beedie You might be interested in this :) This will work well with `str.find` as you mentioned in discord, we can do something like `s = pl.Series(["abc12de"])` `s.str.slice(s.str.find(r"[0-9]+"))`then.

Powerful combination! 😎 
(And will help a lot with some additional SQL support)

File: crates/polars-sql/src/functions.rs [hunk 1]
```diff
@@ -905,11 +905,11 @@ impl SQLFunctionVisitor<'_> {
             Reverse => self.visit_unary(|e| e.str().reverse()),
             Right => self.try_visit_binary(|e, length| {
                 Ok(e.str().slice( match length {
-                    Expr::Literal(LiteralValue::Int64(n)) => -n,
+                    Expr::Literal(LiteralValue::Int64(n)) => lit(-n),
```
[the remark this answers is not in the document]
Comment (alexander-beedie) [reply crates/polars-sql/src/functions.rs:908]: Will do; thx! :))

**Review by ritchie46 — COMMENTED** [review]
Nice one. Thanks @mcrumiller and @reswqa for the unified effort. :smile:

**Review by ritchie46 — APPROVED** [review]

**Comment by mcrumiller** [comment]
@ritchie46 I think you meant to thank @cmdlineluser, I didn't contribute to this one!

**Comment by 9SMTM6** [comment]
I think this silently at the very least broke the API in a release that according to semver should not break API (if polars upholds semver, wasn't able to find a statement at a quick glance).

Luckily I had some assertions going around.

Issue with minimum example coming up.

**Pull request merged**